pilot/pkg/networking/plugin/authz/rbac_test.go

```diff
@@ -529,3 +529,278 @@ func TestConvertRbacRulesToFilterConfig(t *testing.T) {
+			Spec: &rbacproto.ServiceRole{
+				Rules: []*rbacproto.AccessRule{
+					{
+						Services: []string{"service"},
+						Methods:  []string{"GET"},
+					},
+				},
+			},
+		},
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-2"},
+			Spec: &rbacproto.ServiceRole{
+				Mode: rbacproto.EnforcementMode_PERMISSIVE,
+				Rules: []*rbacproto.AccessRule{
+					{
+						Services: []string{"service"},
+						Methods:  []string{"POST"},
+					},
+				},
+			},
+		},
+	}
+	bindings := []model.Config{
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-binding-1"},
+			Spec: &rbacproto.ServiceRoleBinding{
+				Subjects: []*rbacproto.Subject{
+					{
+						User: "user1",
+					},
+				},
+				RoleRef: &rbacproto.RoleRef{
+					Kind: "ServiceRole",
+					Name: "service-role-1",
+				},
+			},
+		},
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-binding-2"},
+			Spec: &rbacproto.ServiceRoleBinding{
+				Mode: rbacproto.EnforcementMode_PERMISSIVE,
+				Subjects: []*rbacproto.Subject{
+					{
+						User: "user2",
+					},
+				},
+				RoleRef: &rbacproto.RoleRef{
+					Kind: "ServiceRole",
+					Name: "service-role-1",
+				},
+			},
+		},
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-binding-3"},
+			Spec: &rbacproto.ServiceRoleBinding{
+				Subjects: []*rbacproto.Subject{
+					{
+						User: "user3",
+					},
+				},
+				RoleRef: &rbacproto.RoleRef{
+					Kind: "ServiceRole",
+					Name: "service-role-2",
+				},
+			},
+		},
+		{
+			ConfigMeta: model.ConfigMeta{Name: "service-role-binding-4"},
+			Spec: &rbacproto.ServiceRoleBinding{
+				Mode: rbacproto.EnforcementMode_PERMISSIVE,
+				Subjects: []*rbacproto.Subject{
+					{
+						User: "user4",
+					},
+				},
+				RoleRef: &rbacproto.RoleRef{
+					Kind: "ServiceRole",
+					Name: "service-role-2",
+				},
+			},
+		},
+	}
+
+	enforcePolicy1 := &policy.Policy{
+		Permissions: []*policy.Permission{{
+			Rule: &policy.Permission_AndRules{
+				AndRules: &policy.Permission_Set{
+					Rules: []*policy.Permission{
+						{
+							Rule: &policy.Permission_OrRules{
+								OrRules: &policy.Permission_Set{
+									Rules: []*policy.Permission{
+										{
+											Rule: &policy.Permission_Header{
+												Header: &route.HeaderMatcher{
+													Name: ":method",
+													HeaderMatchSpecifier: &route.HeaderMatcher_ExactMatch{
+														ExactMatch: "GET",
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		},
+		Principals: []*policy.Principal{{
+			Identifier: &policy.Principal_AndIds{
+				AndIds: &policy.Principal_Set{
+					Ids: []*policy.Principal{
+						{
+							Identifier: &policy.Principal_Authenticated_{
+								Authenticated: &policy.Principal_Authenticated{
+									Name: "user1",
+								},
+							},
+						},
+					},
+				},
+			},
+		}},
+	}
+
+	permissivePolicy1 := &policy.Policy{
```

Comment:
Can you add comment and/or simple function to generate this policy, so it's easier to understand what is the expected output.
Alternative, in the actual test comparison, compare the key component with golden value.

Reply:
updated to use helper functions.